src/components/BottomSheet.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  BottomSheetBackdrop,
+  BottomSheetBackdropProps,
+  BottomSheetModal,
+  BottomSheetView,
+} from "@gorhom/bottom-sheet";
+import Icon from "components/sds/Icon";
+import { Text } from "components/sds/Typography";
+import { getColor } from "config/colors";
+import { THEME } from "config/theme";
+import { pxValue } from "helpers/dimensions";
+import { colorScheme } from "nativewind";
+import React, { useCallback } from "react";
+import { TouchableOpacity, View } from "react-native";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+
+const Icons = {
+  Assets: "Assets",
+  Announcement: "Announcement",
+  Danger: "Danger",
+  Wallet: "Wallet",
+};
+
+type BottomSheetProps = {
+  title: string;
+  description: string;
+  modalRef: React.RefObject<BottomSheetModal | null>;
+  handleCloseModal?: () => void;
+  icon?: keyof typeof Icons;
+  customContent?: React.ReactNode;
+};
+
+const BottomSheet: React.FC<BottomSheetProps> = ({
+  title,
+  description,
+  modalRef,
+  handleCloseModal,
+  icon,
+  customContent,
+}) => {
+  const theme = colorScheme.get() ?? "dark";
+
+  const MapIcons = {
+    [Icons.Assets]: {
+      IconComponent: Icon.Coins01,
+      iconColor: getColor(theme, "mint", 9),
+      backgroundColorClassName: "bg-mint-3",
+    },
+    [Icons.Announcement]: {
+      IconComponent: Icon.Announcement01,
+      iconColor: getColor(theme, "lime", 9),
+      backgroundColorClassName: "bg-lime-3",
+    },
+    [Icons.Danger]: {
+      IconComponent: Icon.AlertTriangle,
+      iconColor: getColor(theme, "red", 9),
+      backgroundColorClassName: "bg-red-3",
+    },
+    [Icons.Wallet]: {
+      IconComponent: Icon.Wallet01,
+      iconColor: getColor(theme, "gold", 9),
+      backgroundColorClassName: "bg-gold-3",
+    },
+  };
+  const IconData = icon ? MapIcons[icon] : null;
+  const insets = useSafeAreaInsets();
+
+  const renderBackdrop = useCallback(
+    (props: BottomSheetBackdropProps) => (
+      <BottomSheetBackdrop
+        {...props}
+        disappearsOnIndex={-1}
+        appearsOnIndex={0}
+        opacity={0.7}
+      />
+    ),
+    [],
+  );
+
+  const renderHandle = useCallback(
+    () => (
+      <View className="bg-background-primary w-full items-center justify-center pt-2 rounded-t-3xl rounded-tr-3xl">
+        <View className="h-1.5 w-10 rounded-full bg-border-primary opacity-35" />
```

Comment:
I don't think it needs to be done in this PR, but we'll need to figure out the best way of incorporating `px()` and `pxValue()` with tailwind